pkg/kubelet/kubelet.go

```diff
@@ -1989,15 +1989,6 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 		return false, nil
 	}
 
-	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && isPodResizeInProgress(pod, &apiPodStatus) {
-		// While resize is in progress, periodically call PLEG to update pod cache
-		runningPod := kubecontainer.ConvertPodStatusToRunningPod(kl.getRuntime().Type(), podStatus)
-		if err, _ := kl.pleg.UpdateCache(&runningPod, pod.UID); err != nil {
```

Comment:
The only purpose of this logic was to update the cache. However, `UpdateCache` underneath invoke the `runtime.GetPodStatus()` which retrieves the latest CRI status, then the cache object stores the latest state which can not be used for state comparison in future `Relist()` loop.